File: app/src/main/java/com/example/octodroid/activities/SearchResultActivity.java
package com.example.octodroid.activities;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.SearchView;
import android.text.TextUtils;
import android.view.Menu;
import android.view.MenuItem;

import com.example.octodroid.R;
import com.example.octodroid.adapters.SearchResultAdapter;

import butterknife.ButterKnife;
import butterknife.InjectView;

public class SearchResultActivity extends ActionBarActivity {
    @InjectView(R.id.repository_list)
    RecyclerView searchResultListView;

    private SearchResultAdapter searchResultAdapter;
    private SearchView.OnQueryTextListener onQueryTextListener = new SearchView.OnQueryTextListener() {
        @Override
        public boolean onQueryTextSubmit(String query) {
            if (!TextUtils.isEmpty(query)) {
                searchResultAdapter.submit(query);
            }
            return false;
        }

        @Override
        public boolean onQueryTextChange(String newText) {
            return false;
        }
    };

    public static void launch(Context context) {
        Intent intent = new Intent(context, SearchResultActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search_result);
        ButterKnife.inject(this);
        setupActionBar();
        setupViews();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_search_result, menu);

        MenuItem menuItem = menu.findItem(R.id.action_search);
        SearchView searchView = (SearchView) menuItem.getActionView();
        searchView.setIconifiedByDefault(true);
        searchView.setSubmitButtonEnabled(false);
        searchView.setQueryHint(getString(R.string.repositories));
        searchView.setOnQueryTextListener(onQueryTextListener);

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            case R.id.action_search:
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void setupActionBar() {
        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setDisplayShowHomeEnabled(true);
    }

    private void setupViews() {
        searchResultAdapter = new SearchResultAdapter(searchResultListView);
        searchResultListView.setAdapter(searchResultAdapter);
    }
}

File: app/src/main/java/com/example/octodroid/activities/SearchResultActivity.kt
package com.example.octodroid.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.SearchView
import android.text.TextUtils
import android.view.Menu
import android.view.MenuItem

import com.example.octodroid.R
import com.example.octodroid.views.adapters.SearchResultAdapter

import butterknife.Bind
import butterknife.ButterKnife

class SearchResultActivity : AppCompatActivity() {
    @Bind(R.id.repository_list)
    internal var searchResultListView: RecyclerView

    private var searchResultAdapter: SearchResultAdapter? = null
    private val onQueryTextListener = object : SearchView.OnQueryTextListener {
        override fun onQueryTextSubmit(query: String): Boolean {
            if (!TextUtils.isEmpty(query)) {
                searchResultAdapter!!.submit(query)
            }
            return false
        }

        override fun onQueryTextChange(newText: String): Boolean {
            return false
        }
    }

    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search_result)
        ButterKnife.bind(this)

        setupActionBar()
        setupViews()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        getMenuInflater().inflate(R.menu.menu_search_result, menu)

        val menuItem = menu.findItem(R.id.action_search)
        val searchView = menuItem.getActionView() as SearchView
        searchView.setIconifiedByDefault(true)
        searchView.setSubmitButtonEnabled(false)
        searchView.setQueryHint(getString(R.string.title_search_result))
        searchView.setOnQueryTextListener(onQueryTextListener)

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.getItemId()) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
            R.id.action_search -> return true
            else -> return super.onOptionsItemSelected(item)
        }
    }

    private fun setupActionBar() {
        val actionBar = getSupportActionBar()
        actionBar.setTitle("")
        actionBar.setDisplayHomeAsUpEnabled(true)
        actionBar.setDisplayShowHomeEnabled(true)
    }

    private fun setupViews() {
        searchResultAdapter = SearchResultAdapter(searchResultListView)
        searchResultListView.setAdapter(searchResultAdapter)
    }

    companion object {

        fun launch(context: Context) {
            val intent = Intent(context, SearchResultActivity::class.java)
            context.startActivity(intent)
        }
    }
}
